Add tracing points for incremental sweeps and opportunistically scheduled tasks

rniwa opened the pull request:
#### df3bb0c1f246089a55bd70fa6e95b61a88850f63
<pre>
Add tracing points for incremental sweeps and opportunistically scheduled tasks
<a href="https://bugs.webkit.org/show_bug.cgi?id=259877">https://bugs.webkit.org/show_bug.cgi?id=259877</a>

Reviewed by Keith Miller and Cameron McCormack.

Added trace points for incremental sweeps and opportunistically scheduled tasks.
This will help us investigate why 266621@main introduced a ~1% regression in Speedometer 2.1.

* Source/JavaScriptCore/heap/IncrementalSweeper.cpp:
(JSC::IncrementalSweeper::doSweep):
* Source/WTF/wtf/SystemTracing.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::performOpportunisticallyScheduledTasks):
* Source/WebKit/Resources/Signposts/SystemTracePoints.plist:

Canonical link: <a href="https://commits.webkit.org/266652@main">https://commits.webkit.org/266652@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3cc61537d13a95728c2b3b64851e81c302ff49a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14369 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14680 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15023 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16110 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/13604 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14494 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17195 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14758 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14548 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15093 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12196 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16829 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/12378 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/12956 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/12297 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/13455 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13120 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/16332 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/13660 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13674 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11518 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/14414 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12959 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3734 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17297 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/14803 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1715 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13519 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3551 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

heycam commented on Source/WTF/wtf/SystemTracing.h:
```diff
@@ -52,6 +52,8 @@ enum TracePointCode {
     DumpJITMemoryStop,
     FromJSStart,
     FromJSStop,
```
Pre-existing, but I noticed these don't have entries in the plist file. Should they, to make it easier for people using Ariadne to profile one off pages that use tracePointStart/tracePointEnd?

rniwa replied:
I think those are exposed as "JSGlobalObject signpost".

heycam reviewed (approved):
r=me but I would prefer if a JSC person could sign off on adding the trace point in IncrementalSweeper (even though you are guarding it behind useTracePoints).

kmiller68 commented on Source/JavaScriptCore/heap/IncrementalSweeper.cpp:
```diff
@@ -74,6 +75,10 @@ void IncrementalSweeper::doWork(VM& vm)
 void IncrementalSweeper::doSweep(VM& vm, MonotonicTime deadline, SweepTrigger trigger)
 {
+    std::optional<TraceScope> traceScope;
+    if (Options::useTracePoints())
```
Can we mark this UNLIKELY?

rniwa replied:
Sure, will do.

kmiller68 reviewed (approved):
r=me with comment

rniwa committed:
Add tracing points for incremental sweeps and opportunistically scheduled tasks
https://bugs.webkit.org/show_bug.cgi?id=259877

Reviewed by Keith Miller and Cameron McCormack.

Added trace points for incremental sweeps and opportunistically scheduled tasks.
This will help us investigate why 266621@main introduced a ~1% regression in Speedometer 2.1.

* Source/JavaScriptCore/heap/IncrementalSweeper.cpp:
(JSC::IncrementalSweeper::doSweep):
* Source/WTF/wtf/SystemTracing.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::performOpportunisticallyScheduledTasks):
* Source/WebKit/Resources/Signposts/SystemTracePoints.plist:

Canonical link: https://commits.webkit.org/266652@main
